pkg/rpcmiddleware/rpclog/server_interceptors.go

```diff
@@ -36,11 +37,22 @@ func UnaryServerInterceptor(ctx context.Context, opts ...Option) grpc.UnaryServe
 		startTime := time.Now()
 		resp, err := handler(newCtx, req)
 
-		if err == nil {
-			if _, ok := o.ignoreMethods[info.FullMethod]; ok {
-				return resp, err
+		if methodLogCfg, ok := o.ignoreMethods[info.FullMethod]; ok {
+			// There is a log configuration for this method.
+			if err != nil {
+				wrapped, ok := errors.From(err)
+				if ok && methodLogCfg.shouldIgnorError(wrapped) {
```

Comment:
You can do `if wrapped, ok := errors.From(err); ok && cfg.shouldIgnore(wrapped) {`.

Reply:
If done like that, in the case where: `IgnoreSuccess == true` and we have an error that shouldn't be ignored, I believe the check will fail and the log will be suppressed?

The change `if err != nil {` -> `if wrapped, ok := errors.From(err); ok {` should be fine tho. What do you think?

Reply:
I didn't refer to merging the `if`s, just having the second `if` have the declarations merged into its body, since neither `ok` nor `wrapped` are used outside of the second `if`s body.

Reply:
Got it, my bad.